packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -100,23 +100,17 @@ contract TaikoL1 is
     }
 
     /// @notice Proves or contests a block transition.
-    /// @param blockId The index of the block to prove. This is also used to
-    /// select the right implementation version.
     /// @param input An abi-encoded {TaikoData.BlockEvidence} object.
-    function proveBlock(
-        uint64 blockId,
-        bytes calldata input
-    )
-        external
-        nonReentrant
-    {
+    function proveBlock(bytes calldata input) external nonReentrant {
```

Comment:
I'm afraid for this function (not the one in LibProving) , we still need the blockId to route to different impls later if we hard fork.

I'll add it back.